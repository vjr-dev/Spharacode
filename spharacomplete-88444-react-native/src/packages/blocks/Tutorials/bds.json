{
  "bds": {
    "name": "Tutorials",
    "id": "Tutorials",
    "platform": [
      "desktop",
      "mobile"
    ],
    "author": "CMT",
    "summary": null,
    "license": "BSD",
    "version": "1.0.0",
    "pages": [
      {
        "skip-to-login": {
          "hotspots": [
            
            {
              "type": "navigation",
              "id": "btnNavigate",
              "options": [
                "target",
                "navigationTransition",
                "navigationType",
                "enabled"
              ]
            }
          ]
        }
      }
    ]
  }
}
