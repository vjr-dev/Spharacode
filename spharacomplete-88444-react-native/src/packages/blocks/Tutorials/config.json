{
  "id": "Tutorials",
  "verion": "0.0.1",
  "status": "preAlpha",
  "path": "./packages/blocks/Tutorials",
  "assembler-adapters": [
    "TutorialsAdapter"
  ],
  "mobile": {
    "merge-engine-navigation": {},
    "assembler-exports": [
      "Tutorials"
    ],
    "assembler-dependencies": [],
    "catalog-dependencies": [
      "core",
      "Tutorials"
    ],
    "assembler-devDependencies": [],
    "merge-engine-screens": {
      "Tutorials": {
        "file": "Tutorials.tsx",
        "extends": "TutorialsController",
        "configURL": "",
        "validationRulesURL": "",
        "imports": [
          {
            "value": "TutorialsController",
            "path": "./TutorialsController"
          },
          {
            "value": "{Props}",
            "path": "./TutorialsController"
          }
        ]
      }
    }
  }
  
}
