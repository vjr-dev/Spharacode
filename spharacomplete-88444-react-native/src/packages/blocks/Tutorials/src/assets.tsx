export const back1 = require("../assets/image_background-1.png");
export const back2 = require("../assets/image_background-2.png");





export const Welcome1 = require("../assets/image_illustration.png");
export const Welcome2 = require("../assets/image_illustration2.png");
export const Welcome3 = require("../assets/image_illustration3.png");
export const Welcome4 = require("../assets/image_illustration4.png");
export const Welcome5 = require("../assets/image_illustration5.png");

