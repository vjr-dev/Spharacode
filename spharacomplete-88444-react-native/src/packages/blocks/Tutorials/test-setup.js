// test-setup.js
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { NativeModules } from 'react-native';

configure({ adapter: new Adapter() });



NativeModules.RNCAsyncStorage = {
    getItem: jest.fn(),
    setItem: jest.fn(),
    removeItem: jest.fn(),
    mergeItem: jest.fn(),
    clear: jest.fn(),
    getAllKeys: jest.fn(),
    flushGetRequests: jest.fn(),
    multiGet: jest.fn(),
    multiSet: jest.fn(),
    multiRemove: jest.fn(),
    multiMerge: jest.fn(),
    replace: jest.fn()
};

jest.mock('@react-native-async-storage/async-storage', () => ({
    getItem: jest.fn(),
    setItem: jest.fn(),
    removeItem: jest.fn(),
    mergeItem: jest.fn(),
    clear: jest.fn(),
    getAllKeys: jest.fn(),
    flushGetRequests: jest.fn(),
    multiGet: jest.fn(),
    multiSet: jest.fn(),
    multiRemove: jest.fn(),
    multiMerge: jest.fn(),
  }))
  


jest.mock('react-native-app-intro-slider', () => {
    const MockedAppIntroSlider = () => <></>;
    return MockedAppIntroSlider;
  });