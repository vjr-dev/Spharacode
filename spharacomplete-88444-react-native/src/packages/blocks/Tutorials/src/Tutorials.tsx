// Customizable Area Start
//@ts-nocheck
//@ts-ignore

import React from "react";

import {
  StyleSheet,
  Text,
  View,
  Platform,
  
  Image,
  ImageBackground,
  Dimensions,
  TouchableOpacity,
  StatusBar
  
} from "react-native";

import TutorialsController, { Props } from "./TutorialsController";


import { back1, back2 } from "./assets";
const windowWidth = Dimensions.get("window").width;
const windowHeight = Dimensions.get("window").height;



import AppIntroSlider from "react-native-app-intro-slider";


export default class Tutorials extends TutorialsController {
  constructor(props: Props) {
    super(props);
  }
  renderItem = ({ item }) => (
    <View style={styles.child}>
      <StatusBar
        animated={true}
        backgroundColor="#454545"
        barStyle={"light-content"}
        showHideTransition={"slide"}
      />
      <ImageBackground source={back1} style={styles.image}>
        <ImageBackground source={back2} style={styles.image2}>
          <Image
            resizeMode="contain"
            source={item?.image}
            style={styles.images}
          />
          <Text style={styles.text}>{item?.text}</Text>
          <TouchableOpacity testID="btnclickID" onPress={() => this.onclick()}>
            <Text style={styles.text2}>SKIP</Text>
          </TouchableOpacity>
        </ImageBackground>
      </ImageBackground>
    </View>
  )

  render() {
    return (
      
      <AppIntroSlider
        dotStyle={{
          backgroundColor: "#3f3f3f",
          bottom:
            Platform.OS === "ios" ? windowHeight - 125 : windowHeight - 100
        }}
        activeDotStyle={{
          backgroundColor: "#f07233",
          bottom:
            Platform.OS === "ios" ? windowHeight - 125 : windowHeight - 100
        }}
        showNextButton={false}
        showDoneButton={false}
        testID='AppintroSliderID'
        renderItem={(item) => (this.renderItem(item))}
        data={this.state.data}
      />
      
    );
  }
}



const styles = StyleSheet.create({
  container: {
    flex: 1,
    height: windowHeight,
    width: windowWidth,
    backgroundColor: "#ffffffff"
  },
  image: {
    height: windowHeight,
    width: windowWidth
  },
  image2: {
    height: windowHeight,
    width: windowWidth,
    alignItems: "center",
    justifyContent: "center"
  },
  container1: { flex: 1, backgroundColor: "white" },
  child: { width: windowWidth },
  pagination: { top: 10 },
  images: { height: 150, width: 150, alignSelf: "center", marginBottom: 40 },
  text: {
    fontSize: 20,
    textAlign: "center",
    color: "#fff",
    marginTop: 20,
    height: 150,
    width: 250
  },
  text2: { fontSize: 15, textAlign: "center", color: "#f07233", marginTop: 50 },
  pdata: { height: 8, width: 8, borderRadius: 4 }
});
// Customizable Area End
