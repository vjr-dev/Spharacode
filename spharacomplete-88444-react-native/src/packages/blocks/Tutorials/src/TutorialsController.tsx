// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import { IBlock } from "../../../framework/src/IBlock";
import { BlockComponent } from "../../../framework/src/BlockComponent";

import { runEngine } from "../../../framework/src/RunEngine";
import AsyncStorage from '@react-native-async-storage/async-storage'



import { Welcome1, Welcome2, Welcome3, Welcome4, Welcome5 } from "./assets";



export const configJSON = require("./config");


export interface Props {
    navigation: any;
    id: string;
    
    
}

interface S {
    
    data: any
    
}

interface SS {
    id: any;
    
    
}

export default class TutorialsController extends BlockComponent<Props, S, SS> {
    
    

    constructor(props: Props) {
        super(props);
        this.receive = this.receive.bind(this);

        
        this.subScribedMessages = [
            
            
        ];

        this.state = {
            
            // data: [1, 2, 3, 4, 5]
            data: [
                { image: Welcome1, text: 'Often feel vulnerable when you are alone ?' },
                { image: Welcome2, text: 'Ensure your safety in unsafe circumstances by ON-HAND emergency Button at all times.' },
                { image: Welcome3, text: 'SHAKE IT gesture to save you from the hassle of unlocking.' },
                { image: Welcome4, text: 'Can integrate with the social media applications to broaden your security network.' },
                { image: Welcome5, text: "Use voice recognition to contact help even when you can't reach your phone. " },
            ]
            
        };
        runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);

        
        
    }

    onclick() {
        AsyncStorage.setItem("Tutorial", "yes")
        this.props.navigation.replace("LoginScreen")
    }
}
// Customizable Area End