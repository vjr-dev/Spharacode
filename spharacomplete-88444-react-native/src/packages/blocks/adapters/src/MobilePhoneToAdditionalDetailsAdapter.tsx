import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class MobilePhoneToAdditionalDetailsAdapter {
  send: (message: Message) => void;

  constructor() {
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.NavigationMobilePhoneAdditionalDetailsMessage)
    ]);
  }

  convert = (from: Message): Message => {
    const to = new Message(getName(MessageEnum.NavigationMessage));

    to.addData(
      getName(MessageEnum.NavigationTargetMessage),
      'AdditionalDetailForm'
    );

    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );

    const raiseMessage: Message = new Message(
      getName(MessageEnum.NavigationPayLoadMessage)
    );
    raiseMessage.addData(
      getName(MessageEnum.AuthTokenDataMessage),
      from.getData(getName(MessageEnum.AuthTokenDataMessage))
    );

    to.addData(getName(MessageEnum.NavigationRaiseMessage), raiseMessage);

    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
