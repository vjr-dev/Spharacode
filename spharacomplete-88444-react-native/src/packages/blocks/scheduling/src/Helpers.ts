/* App/Lib/GeneralHelpers.js */
import { Platform, StyleSheet } from 'react-native';
import { heightFromPercentage } from '../../../framework/src/Utilities';

// Colors Constants
export const COLORS = {
  TRANSPARENT: 'transparent',
  white: 'rgb(255,255,255)',
  black: 'rgb(0,0,0)',
  black10: 'rgb(10,10,10)',
  green: 'rgb(76,217,100)',
  red: 'red',
  yellow: 'rgba(255, 212, 40, 1)',
  darkGray: 'rgba(105,105,105,0.9)',
  lightGray: '#a7b6b9',
  skipGray: 'rgb(91,91,92)',
  borderColor: 'rgb(240, 240, 240)',
  inactiveIndicator: 'rgb(228, 233, 242)',
  inputLabel: 'rgb(143,155,179)',
  inputIcon: 'rgb(197,206,224)',
  inputValue: 'rgb(46,58,89)',
  discountRed: 'rgb(199,5,28)',
  orange: 'rgb(255,157,43)',
  infoGray: 'rgb(163,164,165)',
  ReviewBlack: 'rgb(34,43,69)',
  blue: 'blue',
  blueGrey: 'rgb(143,155,179)',
  darkBlueGrey: 'rgb(46,58,89)',
  lightBlueGrey: 'rgb(197, 206, 224)'
};

export const CommonStyle = StyleSheet.create({
  container: {
    flex: 1,
    width: "100%",
    justifyContent: "flex-start",
    alignItems: "center"
  },
  wrapper: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  headerContainer:{
    backgroundColor:COLORS.yellow,
    height: '10%',
  },
  authHeader:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.yellow
  },
  authHeaderWhite:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.white
  },
  formContainer: {
    width: "80%"
  },
  textAlignCenter:{
    textAlign: 'center'
  },
  labelTextStyle: {
    color: COLORS.inputLabel
  },
  inputContainerStyle:{
    borderBottomWidth: 1,
    borderBottomColor: COLORS.inactiveIndicator
  },
  error: {
    color: COLORS.red,
    fontSize: 14
  },
  scrollStyle: {
    width: "100%"
  },
  btnTitleStyle:{
    fontSize: 17,
    color: COLORS.black
  },
  closeButtonStyle: {
    backgroundColor: COLORS.inputIcon,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  yellowButtonStyle: {
    backgroundColor: COLORS.yellow,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  closeBtnTitleStyle:{
    color: COLORS.black
  },
  scrollContainer: {
    flexGrow: 1,
    alignItems: "center"
  },
  shadowStyle: {
    shadowOffset:
      Platform.OS === "ios"
        ? { width: 0, height: 1 }
        : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 3 : 5,
    elevation: 5
  },
  boxShadowStyle: {
    backgroundColor: COLORS.white,
    marginHorizontal:2,
    marginVertical:1,
    shadowOffset:
    Platform.OS === "ios"
      ? { width: 0, height: 1 }
      : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 1 : 5,
    elevation: 2,    
    borderRadius:8,
  },
  toastModal: {
    borderRadius: 20,
    marginHorizontal: 15,
    paddingHorizontal: '12%',
    paddingVertical: 100,
    alignContent: "center",
    justifyContent: "center",
  },
  clearModal: {
    borderRadius: 5,
    marginHorizontal: 50,
    height: 220,
    backgroundColor: COLORS.red
  },
  imageThumbContain: {
    resizeMode: "contain",
    width: "100%",
    height: "100%"
  },
  imageThumbCover: {
    resizeMode: "cover",
    width: "100%",
    height: "100%"
  },
  flexContainer: {
    flex: 1,
    width: "100%"
  },
  rowStyle: {
    flexDirection: "row"
  },
  flexStart: {
    justifyContent: "flex-start"
  },
  flexEnd: {
    justifyContent: "flex-end"
  },
  jCCenter: {
    justifyContent: "center"
  },
  spaceBetween: {
    justifyContent: "space-between",    
  },
  spaceAround: {
    justifyContent: 'space-around'
  },
  alignItemsCenter:{
    alignItems: "center"
  },
  styleCenter: {
    justifyContent: "flex-end",
    alignItems: "center"
  },
  centerStyle: {
    justifyContent: "center",
    alignItems: "center"
  },
  width100P:{
    width: '100%'
  },
  subView: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  emptyList: {
    alignItems: "center",
    marginTop: 100
  },
  emptyListFilter: {
    alignItems: "center",
    justifyContent: "center",
    marginTop: 100
  },
  backIconView: {
    marginVertical: 10,
    marginHorizontal: 5,
    width: 30,
    height: 30,
    justifyContent: "flex-end",
    alignItems: "flex-end"
  },
  crossImage: {
    tintColor: COLORS.red,
    width: "100%",
    height: "100%",
    resizeMode: "contain"
  },
  searchTextInput: {
    width: "80%"
  },
  errorReport: {
    width: "70%",
    color: COLORS.red,
    fontSize: 15,
    marginLeft: "28.8%"
  }
});