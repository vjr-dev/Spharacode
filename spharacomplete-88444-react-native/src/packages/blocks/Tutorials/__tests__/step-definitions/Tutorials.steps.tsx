//@ts-nocheck
//@ts-ignore
import { defineFeature, loadFeature } from "jest-cucumber"
import { shallow, ShallowWrapper } from 'enzyme'

import * as helpers from '../../../../framework/src/Helpers'
import React from "react";
import {View} from "react-native";

import Tutorials from "../../src/Tutorials"


const screenProps = {
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    replace: jest.fn(),
    addListener: jest.fn().mockImplementation((event: any, callback: any) => {
      callback();
    }),
    state: {
      params: {
        data: { name: 'Test Name' },
        userName: 'abc'
      }
    }
  },
  id: "Tutorials"
}

const feature = loadFeature('./__tests__/features/Tutorials-scenario.feature');

defineFeature(feature, (test) => {


  beforeEach(() => {
    jest.resetModules()
    jest.doMock('react-native', () => ({ Platform: { OS: 'web' } }))
    jest.spyOn(helpers, 'getOS').mockImplementation(() => 'web');
  });
  test('User navigates to Tutorials', ({ given, when, then }) => {
    let tutorialBlock: ShallowWrapper;
    let instance: Tutorials;

    given('I am a User loading Tutorials', () => {
      //@ts-ignore
      tutorialBlock = shallow(<Tutorials {...screenProps} />)
    });

    when('I navigate to the Tutorials', () => {
      instance = tutorialBlock.instance() as Tutorials
    });

    then('Tutorials will load with out errors', () => {
      expect(tutorialBlock).toBeTruthy()
      expect(tutorialBlock).toMatchSnapshot()
      instance.componentDidMount()
      const userData = {data: "data"}
      expect(userData).toBeTruthy()
      const tutorial = {data: "data"}
      expect(tutorial).toBeTruthy()
      
      instance.onclick()
    });
    then('I should able to click the button without error', () => {
      
      let appSlider = tutorialBlock.findWhere(
        (node) => node.prop("testID") === "AppintroSliderID");

      const item = appSlider.renderProp("renderItem")({
        item: {
          image: {
            "testUri": "../../../packages/blocks/Tutorials/assets/image_illustration.png",
          },
          text: "Often feel vulnerable when you are alone ?",
        },
      });
      expect(item).toMatchSnapshot();


      let listWrapper = shallow(
        <View>
          {instance.renderItem(
            {
              id: 3877,
              date: "07-February-23",
              time: "10:48AM",
              alert_type: "PanicIncident",
              address:
                "Varanasi, Sadar, Varanasi, Uttar Pradesh, 221010, India",
              image: "Pending",
            },
            0
          )}
        </View>
      );
      let onPressbtn = listWrapper.findWhere(
        (node) => node.prop("testID") === "btnclickID"
      );
      onPressbtn.simulate("press",instance.onclick());
      expect(onPressbtn).toBeTruthy();

    })
  })
}
)
