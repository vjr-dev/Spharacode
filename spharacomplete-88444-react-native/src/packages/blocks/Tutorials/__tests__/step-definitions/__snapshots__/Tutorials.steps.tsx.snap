// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tutorials User navigates to Tutorials 1`] = `
<MockedAppIntroSlider
  activeDotStyle={
    Object {
      "backgroundColor": "#f07233",
      "bottom": 1209,
    }
  }
  data={
    Array [
      Object {
        "image": Object {
          "testUri": "../../../packages/blocks/Tutorials/assets/image_illustration.png",
        },
        "text": "Often feel vulnerable when you are alone ?",
      },
      Object {
        "image": Object {
          "testUri": "../../../packages/blocks/Tutorials/assets/image_illustration2.png",
        },
        "text": "Ensure your safety in unsafe circumstances by ON-HAND emergency Button at all times.",
      },
      Object {
        "image": Object {
          "testUri": "../../../packages/blocks/Tutorials/assets/image_illustration3.png",
        },
        "text": "SHAKE IT gesture to save you from the hassle of unlocking.",
      },
      Object {
        "image": Object {
          "testUri": "../../../packages/blocks/Tutorials/assets/image_illustration4.png",
        },
        "text": "Can integrate with the social media applications to broaden your security network.",
      },
      Object {
        "image": Object {
          "testUri": "../../../packages/blocks/Tutorials/assets/image_illustration5.png",
        },
        "text": "Use voice recognition to contact help even when you can't reach your phone. ",
      },
    ]
  }
  dotStyle={
    Object {
      "backgroundColor": "#3f3f3f",
      "bottom": 1209,
    }
  }
  renderItem={[Function]}
  showDoneButton={false}
  showNextButton={false}
  testID="AppintroSliderID"
/>
`;

exports[`Tutorials User navigates to Tutorials 2`] = `
<View
  style={
    Object {
      "width": 750,
    }
  }
>
  <StatusBar
    animated={true}
    backgroundColor="#454545"
    barStyle="light-content"
    showHideTransition="slide"
  />
  <ImageBackground
    source={
      Object {
        "testUri": "../../../packages/blocks/Tutorials/assets/image_background-1.png",
      }
    }
    style={
      Object {
        "height": 1334,
        "width": 750,
      }
    }
  >
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/Tutorials/assets/image_background-2.png",
        }
      }
      style={
        Object {
          "alignItems": "center",
          "height": 1334,
          "justifyContent": "center",
          "width": 750,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../packages/blocks/Tutorials/assets/image_illustration.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "height": 150,
            "marginBottom": 40,
            "width": 150,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 20,
            "height": 150,
            "marginTop": 20,
            "textAlign": "center",
            "width": 250,
          }
        }
      >
        Often feel vulnerable when you are alone ?
      </Text>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        testID="btnclickID"
      >
        <Text
          style={
            Object {
              "color": "#f07233",
              "fontSize": 15,
              "marginTop": 50,
              "textAlign": "center",
            }
          }
        >
          SKIP
        </Text>
      </TouchableOpacity>
    </ImageBackground>
  </ImageBackground>
</View>
`;
